src/go/k8s/pkg/resources/configmap.go

```diff
@@ -159,75 +135,43 @@ func (r *ConfigMapResource) Obj() (k8sclient.Object, error) {
 	return cm, nil
 }
 
-func (r *ConfigMapResource) createConfiguration() interface{} {
-	serviceAddress := r.pandaCluster.Name + "." + r.pandaCluster.Namespace + ".svc.cluster.local"
+func (r *ConfigMapResource) createConfiguration() *config.Config {
+	serviceAddress := r.svc.HeadlessServiceFQDN()
 	cfgRpk := config.Default()
 
 	c := r.pandaCluster.Spec.Configuration
-	cr := cfgRpk.Redpanda
-
-	if !r.polymorphicAdvertisedAPI {
-		defaultKafkaAPIPort := 0
-		if len(cfgRpk.Redpanda.KafkaApi) > 0 {
-			defaultKafkaAPIPort = cfgRpk.Redpanda.KafkaApi[0].Port
-		}
+	cr := &cfgRpk.Redpanda
 
-		return &Config{
-			ConfigFile:	configPath,
-			Redpanda: RedpandaConfig{
-				RPCServer: config.SocketAddress{
-					Address:	"0.0.0.0",
-					Port:		getPort(c.RPCServer.Port, cr.RPCServer.Port),
-				},
-				AdvertisedRPCAPI: &config.SocketAddress{
-					Address:	"0.0.0.0",
-					Port:		getPort(c.RPCServer.Port, cr.RPCServer.Port),
-				},
-				KafkaAPI: config.SocketAddress{
-					Address:	"0.0.0.0",
-					Port:		getPort(c.KafkaAPI.Port, defaultKafkaAPIPort),
-				},
-				AdvertisedKafkaAPI: config.SocketAddress{
-					Address:	"0.0.0.0",
-					Port:		getPort(c.KafkaAPI.Port, defaultKafkaAPIPort),
-				},
-				AdminAPI: config.SocketAddress{
-					Address:	"0.0.0.0",
-					Port:		getPort(c.AdminAPI.Port, cr.AdminApi.Port),
-				},
-				DeveloperMode:	c.DeveloperMode,
-				ID:		0,
-				SeedServers: []config.SeedServer{
-					{
-						Host: config.SocketAddress{
-							// Example address: cluster-sample-0.cluster-sample.default.svc.cluster.local
-							Address:	r.pandaCluster.Name + "-0." + serviceAddress,
-							Port:		getPort(c.RPCServer.Port, cr.RPCServer.Port),
-						},
-					},
-				},
-				Directory:	dataDirectory,
+	cr.KafkaApi = []config.NamedSocketAddress{
+		{
+			SocketAddress: config.SocketAddress{
+				Address:	"0.0.0.0",
+				Port:		c.KafkaAPI.Port,
 			},
-		}
+			Name:	"Internal",
+		},
 	}
 
 	cr.RPCServer.Port = getPort(c.RPCServer.Port, cr.RPCServer.Port)
-	cr.AdvertisedRPCAPI.Port = getPort(c.RPCServer.Port, cr.AdvertisedRPCAPI.Port)
+	cr.AdvertisedRPCAPI = &config.SocketAddress{
+		Address:	"0.0.0.0",
+		Port:		getPort(c.RPCServer.Port, cr.RPCServer.Port),
```

Comment:
nit: can we rename this method? portOrDefault or ... I don't know. It's hard to understand what is it doing from getPort and two ports passed into it.

Reply:
Agree will change to `clusterCRPortOrRPKDefault`